### PR TITLE
Refactor initialize of background pool

### LINKED ISSUE #5189: Refactor initialize of background pool
## Enhancement
Now we initialize and get background pool by `Context::getBackgroundPool()` which is kind of confusing, we should separate `initialization` and `get`.

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: close #5189

Problem Summary:

### What is changed and how it works?
Add `Context::initializeBackgroundPool(UInt16 pool_size)` and `Context::initializeBlockableBackgroundPool(UInt16 pool_size)` to initialize background thread pool. And remove check in `Context::getBackgroundPool()` and `Context::getBlockableBackgroundPool()`

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Server/Server.cpp [hunk 1]
```diff
@@ -1402,7 +1400,7 @@ int Server::main(const std::vector<std::string> & /*args*/)
     {
         auto size = settings.grpc_completion_queue_pool_size;
         if (size == 0)
-            size = std::thread::hardware_concurrency();
+            size = server_info.cpu_info.logical_cores;
```
Comment (JaySon-Huang) [inline dbms/src/Server/Server.cpp:1403]: leave the value of `size` unchanged?

Comment (Lloyd-Pottiger) [inline dbms/src/Server/Server.cpp:1403]:
```suggestion
            size = std::thread::hardware_concurrency();
```

Comment (Lloyd-Pottiger) [reply dbms/src/Server/Server.cpp]: done

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Review by JinheLin — APPROVED** [review]

**Pull request merged**